source/dagger/gameplay/team_game/door_interaction.cpp

```diff
@@ -15,52 +15,58 @@ using namespace team_game;
 
 using namespace dagger;
 
-bool keytaken = false;
 
+Bool pickedup = false;
 void DoorSystem::Run()
 {
+
 	
 	
-	auto viewkey= Engine::Registry().view<Transform, SimpleCollision, Key>();
+	auto view = Engine::Registry().view<Detection, SimpleCollision,StaticBody,Door>();
+
+	auto viewkey = Engine::Registry().view<Key>();
 	for (auto keyentity : viewkey)
 	{
 		auto& key = viewkey.get<Key>(keyentity);
 		if (key.GetPickedUp() == true) {
-			keytaken = true;
+			pickedup = true;
 			Engine::Registry().destroy(keyentity);
 		}
+		else { pickedup = false; }
```

Comment:
try to follow some codestyle:
put "{" after new line or not, decide this and then do it in one way. 

I see different ways inside one file